app/src/main/cpp/skyline/services/account/IProfile.cpp

```diff
@@ -53,14 +44,30 @@ namespace skyline::service::account {
     }
 
     Result IProfile::GetImageSize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        response.Push<u32>(profileImageIcon.size());
+        std::shared_ptr<vfs::Backing> profileImageIcon{GetProfilePicture()};
+        response.Push<u32>(profileImageIcon->size);
+
         return {};
     }
 
     Result IProfile::LoadImage(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        // TODO: load actual profile image
-        request.outputBuf.at(0).copy_from(profileImageIcon);
-        response.Push<u32>(profileImageIcon.size());
+        std::shared_ptr<vfs::Backing> profileImageIcon{GetProfilePicture()};
+        std::vector<char> profileImage(profileImageIcon->size);
+
+        profileImageIcon->Read(skyline::span<char>(profileImage), 0);
+
+        request.outputBuf.at(0).copy_from(profileImage);
```

Comment:
Rather than reading it into `profileImage` first then copying it into the output buffer, you can directly read it into `outputBuf`.